execution/distinct.go

```diff
@@ -13,13 +13,18 @@ type Distinct struct {
 	storage        storage.Storage
 	source         Node
 	eventTimeField octosql.VariableName
+
+	garbageCollectorBoundary int // number of minutes subtracted from watermark to create boundary for garbage collection
+	garbageCollectorCycle    int // number of miliseconds for garbage collector to sleep between records check
 }
 
-func NewDistinct(storage storage.Storage, source Node, eventTimeField octosql.VariableName) *Distinct {
+func NewDistinct(storage storage.Storage, source Node, eventTimeField octosql.VariableName, garbageCollectorBoundary, garbageCollectorCycle int) *Distinct {
```

Comment:
Dunno if you agree, but maybe it would be more elegant to create a `type garbageCollectorInfo struct` that contains both these values and then just pass this to the constructor - I think it looks better tbh